### PR TITLE
Fabo/fix balances not updating

### LINKED ISSUE #4163: Balances from other networks in ActionModal
Very often using the new networks navbar on the left, when we try to make a transaction we will see a wrong validation that doesn't allow us to continue:

The reason for this errors is because the balances in ActionModal don't get properly updated on network change (although they do in SendModal and the rest on the contrary), so `selectedBalance` is searching for a balance with a denom that actually belongs to another network, which returns `defaultBalance` with amount 0 🤷 

The user can only proceed by refreshing the page.

### PR DESCRIPTION
**Pull request by faboweb**
Closes #4163, #4172 

@Bitcoinera check this one please

### CODE REVIEW

File: src/ActionModal/components/ActionModal.vue [hunk 1]
```diff
@@ -779,7 +778,7 @@ export default {
       /* istanbul ignore next */
       update(data) {
         this.balancesLoaded = true
-        return data.balances || []
+        return JSON.parse(JSON.stringify(data.balances)) || []
```
Comment (Bitcoinera) [inline src/ActionModal/components/ActionModal.vue:781]: This works?? 👁️ I'm going to check

Comment (faboweb) [reply src/ActionModal/components/ActionModal.vue:781]: Went on a hunch and worked for me. Assumed change is not triggered by the new data somehow.

Comment (Bitcoinera) [reply src/ActionModal/components/ActionModal.vue:782]: Not working for me.

Could you reproduce the bug?

File: src/main.js [hunk 2]
```diff
@@ -33,6 +34,7 @@ Vue.use(Vuelidate)
 Vue.use(VueClipboard)
 Vue.use(InfiniteScroll)
 Vue.use(AsyncComputed)
+Vue.use(DynamicReactiveRefs)
```
Comment (Bitcoinera) [inline src/main.js:37]: 👀

Comment (faboweb) [reply src/main.js:37]: makes the refs reactive so we can trigger logic on $refs.actionModal being available

**Comment by Bitcoinera** [comment]
Seems fixed 👍 Awesome job!

File: src/main.js (hunk 2 above)
Comment (Bitcoinera) [reply src/main.js:37]: thank you!

File: src/ActionModal/components/UndelegationModal.vue [hunk 3]
```diff
@@ -411,7 +411,7 @@ export default {
       `,
       /* istanbul ignore next */
       skip() {
-        return !this.address
+        return !this.address || !this.$refs.actionModal.show
```
Comment (Bitcoinera) [inline src/ActionModal/components/UndelegationModal.vue:414]:
```suggestion
        return !this.address ||  !this.$refs.actionModal || !this.$refs.actionModal.show
```

File: src/ActionModal/components/UndelegationModal.vue [hunk 4]
```diff
@@ -474,7 +474,7 @@ export default {
       },
       /* istanbul ignore next */
       skip() {
-        return !this.address
+        return !this.address || !this.$refs.actionModal.show
```
[the remark this answers is not in the document]
Comment (Bitcoinera) [reply src/ActionModal/components/UndelegationModal.vue:477]: Or is there a reason why they are not completely consistent?

[the remark this answers is not in the document]
Comment (faboweb) [reply src/ActionModal/components/UndelegationModal.vue]: nice find. mussed have missed it

**Pull request merged**